components/core/src/clp/streaming_archive/writer/Archive.cpp

```diff
@@ -635,8 +634,6 @@ void Archive::update_metadata() {
     m_metadata_file_writer.seek_from_begin(0);
     m_local_metadata->write_to_file(m_metadata_file_writer);
 
-    m_global_metadata_db->update_archive_metadata(m_id_as_string, *m_local_metadata);
-
     if (m_print_archive_stats_progress) {
```

Comment:
This block should probably move into `Archive::close` since it may mislead the external caller if we indicate that an archive's stats have been updated, but that archive doesn't exist in the global metadata database (whereas it did before this change).

Reply:
The progress output is being [used by the package](https://github.com/y-scope/clp/blob/main/components/job-orchestration/job_orchestration/executor/compress/compression_task.py#L322) to keep track of when new archives have been generated. The package expects that the archive exists in the global metadata database when it receives the progress update. Progress is probably a poor name for this option at this point since its use is not really for humans per se.

Reply:
True, it does use only the last update. For the sake of safety, can you add a comment indicating that the stats are printed even though the archive doesn't exist in the global metadata database?

Reply:
> If you want to be more safe. I could potential add a duplicate stat printing when the archive is closed (maybe with a new field like complete) lmk

Good idea. Sgtm.